jrnl/EncryptedJournal.py

```diff
@@ -40,8 +40,11 @@ def open(self, filename=None):
         """Opens the journal file defined in the config and parses it into a list of Entries.
         Entries have the form (date, title, body)."""
         filename = filename or self.config["journal"]
-
+        dirname = os.path.dirname(filename)
         if not os.path.exists(filename):
+            if not os.path.isdir(dirname):
+                os.mkdir(dirname)
```

Comment:
Instead of `os.mkdir`, please us `os.makedirs` in case there are nested directories that need to be created.